File: 1006/B/solution.cpp
#include<bits/stdc++.h>
using namespace std;
const int array_len = 2345;
int n, k; 
pair<int,int> a[array_len];
pair<int,int> b[array_len];
int main() {
	cin>>n>>k;
	for(int i = 0;i<n;i++) cin>>a[i].first, a[i].second = i+1;
	sort(a, a+n);
	int sum = 0;
	for(int i = 1;i<=k;i++) sum += a[n-i].first;
	cout<<sum<<endl;
	for(int i = 1;i<=k;i++) {
		b[i].first = a[n-i].second;
		b[i].second = a[n-i].first;
	}
	sort(b+1, b+k+1);
	for(int i = 1;i<k;i++) cout<<b[i].first - b[i-1].first<<' ';
	cout<<n - b[k-1].first<<endl;
	return 0;
}

File: 1006/B/solution.go
package main

import (
	"fmt"
	"sort"
)

func ReadInt64() int64 {
	var ret int64
	_, err := fmt.Scan(&ret)
	if err != nil {
		return 0
	}
	return ret
}

type pair struct {
	first  int64
	second int64
}

func main() {
	n, k, sum := int64(0), int64(0), int64(0)
	n, k = ReadInt64(), ReadInt64()
	a, b := make([]pair, n), make([]pair, k+1)

	for i := int64(0); i < n; i++ {
		a[i].first = ReadInt64()
		a[i].second = i + 1
	}
	sort.Slice(a, func(i, j int) bool { return a[i].first < a[j].first })
	for i := int64(0); i < k; i++ {
		sum += a[n-i-1].first
	}
	fmt.Printf("%d\n", sum)
	for i := int64(1); i <= k; i++ {
		b[i] = a[n-i]
	}
	sort.Slice(b, func(i, j int) bool { return b[i].second < b[j].second })
	for i := int64(1); i < k; i++ {
		fmt.Printf("%d ", b[i].second-b[i-1].second)
	}
	fmt.Printf("%d\n", n-b[k-1].second)
}

